### ISSUE TITLE
Consul ignores verify* flags if ca_file is set

### ISSUE THREAD

**Issue by goran-rumin** [hashicorp/consul#4818]
#### Overview of the Issue

We decided to change certificates used by Consul, including CA. Version is 1.2.3. For that we disabled outgoing verification on all nodes (other verify flags are at their default values, which is false), and started the change. After first node, we started getting verification errors. 
After removing `ca_file` option, nodes were able to communicate again without TLS. With `ca_file` option set, TLS is used despite all verify flags being set to false. This was confirmed with Wireshark. In my opinion this indicates issue from the title.

#### Reproduction Steps

1. Create a cluster with 3 server nodes
2. Configure all of them with one set of certificates, with disabled verification
3. Change certificates on one node

### Consul info for Server

<details>
  <summary>Server info</summary>

```
/ # consul info -token=<token>
agent:
	check_monitors = 0
	check_ttls = 1
	checks = 1
	services = 1
build:
	prerelease = 
	revision = 48d287ef
	version = 1.2.3
consul:
	bootstrap = false
	known_datacenters = 1
	leader = true
	leader_addr = <address>:8300
	server = true
raft:
	applied_index = 4512266
	commit_index = 4512266
	fsm_pending = 0
	last_contact = 0
	last_log_index = 4512266
	last_log_term = 77
	last_snapshot_index = 4506959
	last_snapshot_term = 77
	latest_configuration = [{Suffrage:Voter ID:da89405a-216e-2c68-ca57-fd55d099c17e Address:<address1>:8300} {Suffrage:Voter ID:afcf4594-9b6b-223f-034a-3b34331e43aa Address:<address2>:8300} {Suffrage:Voter ID:d95bfed5-b3ca-60e5-707a-4cbe94df706f Address:<address3>:8300}]
	latest_configuration_index = 4460624
	num_peers = 2
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Leader
	term = 77
runtime:
	arch = amd64
	cpu_count = 2
	goroutines = 115
	max_procs = 2
	os = linux
	version = go1.10.1
serf_lan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 31
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 35
	members = 3
	query_queue = 0
	query_time = 1
serf_wan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 13
	members = 3
	query_queue = 0
	query_time = 1
```

</details>

### Operating system and Environment details

Deployed in Docker, host uses CentOS 7.3

### Log Fragments

Logs on leader with old certificate:
`2018/10/18 12:40:56 [WARN] consul: error getting server health from "<name>": rpc error getting client: failed to get conn: x509: certificate signed by unknown authority`

Logs on node with new certificate (has a Vault as client):
`2018/10/18 12:41:24 [ERR] http: Request GET /v1/kv/vault/core/lock, error: No cluster leader from=127.0.0.1:49258`

**Comment by banks** [comment]
Thanks for reporting this @goran-rumin that does sound like unexpected behaviour to me.

Can you add your server and client config in the state where you saw the unexpected behaviour? Note that as in the docs if you left `verify_server_hostname` set in config then it _implies_ `verify_outgoing` too. Also note from the code it looks like setting a `server_name` will also cause verification to be enabled: https://github.com/hashicorp/consul/blob/b5abf61963c7b0bdb674602bfb64051f8e23ddb1/tlsutil/config.go#L140-L148

FYI we are working on massive improvements to our TLS setup and part of that will be to make an easier and more secure way to rotate agent CA and avoid this sort of thing. This should still be looked at in the interim though.

**Comment by goran-rumin** [comment]
We have 3 server nodes and no agent client nodes. Configs are below.

<details>
  <summary>Old config (2 nodes, with self-signed certificates)</summary>

```
{
  "server": true,
  "bind_addr": "<ip addr>",
  "addresses": {
    "https": "0.0.0.0"
  },
  "ports": {
    "dns": -1,
    "https": 8443
  },
  "key_file": "/consul/config/key.pem",
  "cert_file": "/consul/config/cert.pem",
  "ca_file": "/consul/config/cert.pem",
  "encrypt": "<key>",
  "disable_remote_exec": true,
  "verify_outgoing": false,
  "datacenter": "DC",
  "node_name": "node1",
  "ui": true,
  "bootstrap_expect": 3,
  "retry_join": [
    "node2",
    "node3"
  ],
  "acl_datacenter": "DC",
  "acl_default_policy": "deny",
  "acl_master_token": "<token>",
  "acl_agent_token": "<token>"
}
```

</details>

<details>
  <summary>New config (1 node, with legit cert)</summary>

```
{
  "server": true,
  "bind_addr": "<ip addr>",
  "addresses": {
    "https": "0.0.0.0"
  },
  "ports": {
    "dns": -1,
    "https": 8443
  },
  "key_file": "/consul/config/key.pem",
  "cert_file": "/consul/config/cert.pem",
  "ca_path": "/etc/pki/ca-trust/source/anchors/",
  "encrypt": "<key>",
  "disable_remote_exec": true,
  "verify_outgoing": false,
  "datacenter": "DC",
  "node_name": "node1",
  "ui": true,
  "bootstrap_expect": 3,
  "retry_join": [
    "node2",
    "node3"
  ],
  "acl_datacenter": "DC",
  "acl_default_policy": "deny",
  "acl_master_token": "<token>",
  "acl_agent_token": "<token>"
}
```

</details>

Certificate is issued for `<hostname>.<internal_domain>` and has IP address as alternative name. CA is present on CA path.

**Comment by banks** [comment]
^ the PR fix for this is still in 1.4.1, just removing it because it's confusing to have two items in there to track for it.
